python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+different catalogs and the DB.
+"""
+
+
+def make_minimal_dia_object_schema(filter_names=[]):
+    """Define and create the minimal schema required for a DIAObject.
+
+    Parameters
+    ----------
+    filter_names : `list` of `str`s
+        Names of the filters expect and compute means for.
+
+    Return
+    ------
+    schema : `lsst.afw.table.Schema`
+        Minimal schema for DIAObjects.
+    """
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    # For the MVP/S we currently only care about the position though
+    # in the future we will add summary computations for fluxes etc.
+    # as well as their errors.
+
+    # In the future we would like to store a covariance of the coordinate.
+    # This functionality is not defined in currently in the stack, so we will
+    # hold off until it is implemented. This is to be addressed in DM-7101.
+    schema.addField("pixelId", type='L')
+    schema.addField("nDiaSources", type='L')
+    for filter_name in filter_names:
+        schema.addField("psFluxMean_%s" % filter_name, type='D')
+        schema.addField("psFluxMeanErr_%s" % filter_name, type='D')
+        schema.addField("psFluxSigma_%s" % filter_name, type='D')
+    return schema
+
+
+def make_minimal_dia_source_schema():
+    """ Define and create the minimal schema required for a DIASource.
+
+    Return
+    ------
+    schema : `lsst.afw.table.Schema`
+        Minimal schema for DIAObjects.
+    """
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    schema.addField("diaObjectId", type='L')
+    schema.addField("ccdVisitId", type='L')
+    schema.addField("psFlux", type='D')
+    schema.addField("psFluxErr", type='D')
+    schema.addField("filterName", type='String', size=10)
+    schema.addField("filterId", type='L')
+    return schema
+
+
+def ccdVisitSchema():
+    """Define the schema for the CcdVisit table.
+
+    Return
+    ------
+    ccdVisitNames: `collections.OrderedDict`
+       Names of columns in the ccdVisit table.
+    """
+    return oDict([("ccdVisitId", "INTEGER PRIMARY KEY"),
+                  ("ccdNum", "INTEGER"),
+                  ("filterName", "TEXT"),
+                  ("filterId", "INTEGER"),
+                  ("ra", "REAL"),
+                  ("decl", "REAL"),
+                  ("expTime", "REAL"),
+                  ("expMidptMJD", "REAL"),
+                  ("fluxZeroPoint", "REAL"),
+                  ("fluxZeroPointErr", "REAL")])
+
+
+def get_ccd_visit_info_from_exposure(exposure):
+    """
+    Extract info on the ccd and visit from the exposure.
+
+    Paramters
+    ---------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to store information from.
+
+    Returns
+    -------
+    values : `dict` of ``values``
+        List values representing info taken from the exposure.
+    """
+    visit_info = exposure.getInfo().getVisitInfo()
+    sphPoint = exposure.getWcs().getSkyOrigin()
+    flux0, flux0_err = exposure.getCalib().getFluxMag0()
```

Comment:
I worry about storing the magnitude 0 point this way: PhotoCalibs will be more complicated, so this will only be an approximation.

Reply:
I guess my response here is that there isn't really a better thing to store right now. The good news is that the database is storing the CccdVisitId which should allow us to look up the complicated calibration object regardless.